datajoint/autopopulate.py

```
@@ -322,6 +338,7 @@ def _populate1(
                     )
                     if jobs is not None:
                         jobs.complete(self.target.table_name, self._job_key(key))
+                    return True
```

Not sure about this `True`. The docstring needs an update to explain. It's a bit odd to have so many different return types.

---

What about returning `True/False` for success/failure instead of None?

---

When does it ever return `False`?

---

Then you should add `return False` to the case when the function returns without populating anything.

---

Does not populating anything constitute a `False`? `False` seems to indicate failure (maybe that's just me thinking so).

---

The current function produces `True` if successful, `None` if nothing is populated, throws an exception, or returns `(key, error)`. It never returns `False`. It seems that if a function returns `True`, it should also return `False` in other cases. Perhaps instead of `None`, we should return `False`. It would not indicate an error, but simply not populating for other reasons than an error.

---

Agreed, I'll update accordingly